cfgov/v1/models/learn_page.py

```diff
@@ -301,20 +304,46 @@ class EventPage(AbstractFilterPage):
     def page_js(self):
         return super(EventPage, self).page_js + ['video-player.js']
 
+    def get_venue_coords(self):
+        # Default to Washington DC coordinates
+        venue_coords = '-77.039628,38.898238'
+
+        if not self.venue_city or not self.venue_state:
+            return venue_coords
+
+        location = '{} {}'.format(self.venue_city, self.venue_state)
+        geocode_api_url = 'https://api.mapbox.com/geocoding/v5/mapbox.places/'
+        geocode_location_url = geocode_api_url + location + '.json'
+
+        params = {'access_token': settings.MAPBOX_ACCESS_TOKEN}
+        response = requests.get(geocode_location_url, params=params)
+
+        if response.status_code != 200:
+            return venue_coords
+
+        try:
+            geo_data = json.loads(response.text)
+            coordinates = geo_data['features'][0]['geometry']['coordinates']
+            venue_coords = str(coordinates[0]) + ',' + str(coordinates[1])
+        except KeyError:
+            pass
+
+        return venue_coords
+
     def location_image_url(self, scale='2', size='276x155', zoom='12'):
-        center = 'Washington, DC'
-        if self.venue_city:
-            center = self.venue_city
-        if self.venue_state:
-            center = center + ', ' + self.venue_state
-        options = {
-            'center': center,
-            'scale': scale,
-            'size': size,
-            'zoom': zoom
-        }
-        url = 'https://maps.googleapis.com/maps/api/staticmap?'
-        return '{url}{options}'.format(
-            url=url,
-            options=urlencode(options)
+        if not self.venue_coords:
+            self.save()
```

Comment:
This call worries me a bit. I don't think it's a good idea to `save()` the page model _at render time_ if the coordinates aren't available. For example, how does this work when you are previewing a new draft page, or a draft revision of an existing page? Will it save a version of the page just because you previewed?

Reply:
Makes sense! I'll remove it and we can write a script to update old events.